### PR TITLE
Put a message in related location messages, even when using snippets

### PR DESCRIPTION
**Pull request by keyboardDrummer**
### Description
Fill in empty messages for related locations

Shows

image

Instead of

image

### How has this been tested?
- Added a language server test

<small>By submitting this pull request, I confirm that my contribution is made under the terms of the [MIT license](https://github.com/dafny-lang/dafny/blob/master/LICENSE.txt).</small>

### CODE REVIEW

File: Source/DafnyCore/Generic/ReporterExtensions.cs [hunk 1]
```diff
@@ -59,6 +59,10 @@ public static class ErrorReporterExtensions {
       }
     }
 
+    if (message == null) {
+      message = "this expression could not be proved";
```
Comment (MikaelMayer) [inline Source/DafnyCore/Generic/ReporterExtensions.cs:63]:
```suggestion
    message ??= "this expression could not be proved";
```

File: Source/DafnyCore/Generic/ReporterExtensions.cs [hunk 2]
```diff
@@ -59,6 +59,10 @@ public static class ErrorReporterExtensions {
       }
     }
 
+    if (message == null) {
```
Comment (MikaelMayer) [inline Source/DafnyCore/Generic/ReporterExtensions.cs:62]:
```suggestion
```

File: Source/DafnyCore/Generic/ReporterExtensions.cs [hunk 3]
```diff
@@ -59,6 +59,10 @@ public static class ErrorReporterExtensions {
...
+      message = "this expression could not be proved";
+    }
```
Comment (MikaelMayer) [inline Source/DafnyCore/Generic/ReporterExtensions.cs:64]:
```suggestion
```

**Comment by keyboardDrummer** [comment]
> This sounds useful, thanks. The term 'expression' is my second favorite for this kind of message, my first favorite would be the term [proposition](https://en.m.wikipedia.org/wiki/Proposition), so "Related position: this proposition could not be proved". Expression are of any type, but propositions are always boolean. Would you be ok with the term "proposition"?

Good idea. I'll change to proposition

> I was a bit worried about the use of the generic "expression" term, especially because we essentially prove an instantiated version of it, but other terms didn't look more appealing to me (sub-expression, conjecture, proposition)

This seems to contradict what you said before. I'm confused.

File: Source/DafnyLanguageServer.Test/Diagnostics/VerificationDiagnostics.cs [hunk 4]
```diff
@@ -13,6 +13,36 @@ namespace Microsoft.Dafny.LanguageServer.IntegrationTest.Synchronization;
 
 public class VerificationDiagnostics : ClientBasedLanguageServerTest {
 
+  [Fact]
+  public async Task SubExpression() {
```
Comment (MikaelMayer) [inline Source/DafnyLanguageServer.Test/Diagnostics/VerificationDiagnostics.cs:17]:
```suggestion
  public async Task SubProposition() {
```

Really a nit pick, feel free to ignore if you can merge earlier.

**Review by MikaelMayer — APPROVED** [review]
Great! It makes Dafny a little more user-friendly.

**Pull request merged**